app/views/unattended/provisioning_templates/PXEGrub2/preseed_default_pxegrub2_autoinstall.erb

```diff
@@ -20,20 +20,13 @@ test_on:
 #
 <%
   efi_suffix = 'efi'
-  image_path = @preseed_path.sub(/\/?$/, '.iso')
-  options = []
-  options << 'ramdisk_size=1500000'
-  options << 'root=/dev/rd/0 rw auto'
-  options << "url=http://#{@preseed_server}#{image_path}"
-  options << 'cloud-config-url=/dev/null'
-  options << 'autoinstall'
 -%>
 
 set default=0
 set timeout=<%= host_param('loader_timeout') || 10 %>
 
 menuentry '<%= template_name %>' {
-  linux<%= efi_suffix %> <%= @kernel %> <%= options.compact.join(' ') %> "ds=nocloud-net;s=http://<%= foreman_request_addr %>/userdata/" <%= snippet('preseed_kernel_options_autoinstall').strip %>
+  linux<%= efi_suffix %> <%= @kernel %> root=/dev/rd/0 rw auto <%= snippet('preseed_kernel_options_autoinstall').strip %>
```

Comment:
I think it is a good idea to put generic options in the `preseed_kernel_options_autoinstall` file. Unfortunately, we experienced issues in doing so for the PXEGrub2 template:

Since there is a semicolon in `..ds=nocloud-net;s=http://..`, it is important for the PXEGrub2 parser to have this part in double quotes - otherwise, it will stop after the semicolon and ignore any other parameter after it. Due to this, it is explicitly excluded in the `options` variable. As you can see [here](https://github.com/theforeman/foreman/pull/9644/files#diff-87874a35a72c8e66235ee63c9009784f31382181a114b88d63e24e74919d29f6L18), adding it to the options variable removes the quotes.
All other parameters seem fine to me.

Do you either have an idea (a) on how to include the double quotes or (b) do you think we should leave the `root=/dev..` and the `..ds=nocloud-net;s=http://..` as an individual option in every template?

Reply:
We can test it. Let's add the quotes to the `preseed_kernel_options_autoinstall` and afterwards test if PXELinux and iPXE parse it correctly still.

I'll add a change to the comment at the specific line *.